### ISSUE TITLE
尝试按照官方步骤转换yolact出现错误

### ISSUE THREAD

**Issue by gujingxiao** [Tencent/ncnn#1706]
如题，我按照官方的讲解 https://zhuanlan.zhihu.com/p/128974102 ，尝试了yolact的转换，转换过程中都比较顺利，但是转换后的结果与官方给的出现了差别，并且在使用ncnn运行中出现了错误，具体描述如下：
系统环境：Ubuntu16.04
                  pytorch1.4.0
                  ncnn版本为最新的20200413  
                  onnx版本 1.6.0
                  onnx simplier 0.2.6
                  onnx runtime 1.2.0

操作步骤：
1. 操作步骤以及使用的模型完全按照官方流程操作，整个转换过程没有出现问题，但是转换出来的param里面层的name与官方给的对应不上，可以参考下面给出的附件[yolact.txt](https://github.com/Tencent/ncnn/files/4502467/yolact.txt)。
    官方example里面给出的取出层分别为：    
       ex.extract("619", maskmaps);// 138x138 x 32
       ex.extract("816", location);// 4 x 19248
       ex.extract("818", mask);// maskdim 32 x 19248
       ex.extract("820", confidence);// 81 x 19248

   我操作之后，对应的层为617   813   815   817

2.  模型转好之后，我使用了example代码进行推理，加载模型等都正常，但是在
       ex.extract("617", maskmaps);// 138x138 x 32
       ex.extract("813", location);// 4 x 19248
       ex.extract("815", mask);// maskdim 32 x 19248
       ex.extract("817", confidence);// 81 x 19248
    运行之后，只有maskmaps里面有数值，location、mask和confidence里面全部为空
![微信截图_20200420180110](https://user-images.githubusercontent.com/13788388/79739577-f3ca1f00-8330-11ea-83ec-f56e02e83584.png)
![微信截图_20200420180200](https://user-images.githubusercontent.com/13788388/79739648-0d6b6680-8331-11ea-9d24-bed2bf75e1dc.png)

3. 所以目前不知道具体哪里操作错误，还请大佬帮忙分析，感谢感谢

**Comment by nihui** [comment]
你的 param 中 Concat axis 没改啊
0x6 手工微调模型 这一段再看看

**Comment by gujingxiao** [comment]
> 你的 param 中 Concat axis 没改啊
> 0x6 手工微调模型 这一段再看看

我刚刚核对的时候也发现，感谢官方答复，现在结果正常了，谢谢！

**Comment by liulai** [comment]
> > 你的 param 中 Concat axis 没改啊
> > 0x6 手工微调模型 这一段再看看
> 
> 我刚刚核对的时候也发现，感谢官方答复，现在结果正常了，谢谢！

老哥，我跑这个yolact.cpp，模型都能跑出结果，但是后面报错了。请问下你用的什么设备啊？

**Comment by piaoliuping123** [comment]
> > 你的 param 中 Concat axis 没改啊
> > 0x6 手工微调模型 这一段再看看
> 
> 我刚刚核对的时候也发现，感谢官方答复，现在结果正常了，谢谢！

你好，dalao, 我遇到和你一样的问题，请问你是怎么解决的？谢谢

> > > 你的 param 中 Concat axis 没改啊
> > > 0x6 手工微调模型 这一段再看看
> > 
> > 
> > 我刚刚核对的时候也发现，感谢官方答复，现在结果正常了，谢谢！
> 
> 老哥，我跑这个yolact.cpp，模型都能跑出结果，但是后面报错了。请问下你用的什么设备啊？

您好，我在执行./ncnnoptimize yolact.param yolact.bin yolact-opt.param yolact-opt.bin 0 这条命令的时候报错，显示如下：
Input layer input.1 without shape info, shape_inference aborted
unsupported resize type 0
unsupported resize type 0
unsupported resize type 0
请问怎么解决，谢谢了

**Comment by zqnnn** [comment]
> > > 你的 param 中 Concat axis 没改啊
> > > 0x6 手工微调模型 这一段再看看
> > 
> > 
> > 我刚刚核对的时候也发现，感谢官方答复，现在结果正常了，谢谢！
> 
> 你好，dalao, 我遇到和你一样的问题，请问你是怎么解决的？谢谢
> 
> > > > 你的 param 中 Concat axis 没改啊
> > > > 0x6 手工微调模型 这一段再看看
> > > 
> > > 
> > > 我刚刚核对的时候也发现，感谢官方答复，现在结果正常了，谢谢！
> > 
> > 
> > 老哥，我跑这个yolact.cpp，模型都能跑出结果，但是后面报错了。请问下你用的什么设备啊？
> 
> 您好，我在执行./ncnnoptimize yolact.param yolact.bin yolact-opt.param yolact-opt.bin 0 这条命令的时候报错，显示如下：
> Input layer input.1 without shape info, shape_inference aborted
> unsupported resize type 0
> unsupported resize type 0
> unsupported resize type 0
> 请问怎么解决，谢谢了

你解决这个问题了吗？我也遇到了

**Comment by nihui** [comment]
Input layer input.1 without shape info, shape_inference aborted
unsupported resize type 0

这两种提示可以无视的...

**Comment by zqnnn** [comment]
> Input layer input.1 without shape info, shape_inference aborted
> unsupported resize type 0
> 
> 这两种提示可以无视的...

忽略这个进行./ncnn2table会出现这个问题。。
./ncnn2table --param ../psenet_lite_nobn_fp32.param --bin ../psenet_lite_nobn_fp32.bin --images ../../../../data/final_test/imgs/ --output psenet_lite_nobn.table --mean=104,117,123 --norm=0.017,0.017,0.017 --size=185,40 --thread=2
--- ncnn post training quantization tool --- 16:26:39 Apr 26 2020
fopen true failed
fopen true failed
not found [Input] Layer, Check your ncnn.param

**Comment by gujingxiao** [comment]
> > > 你的 param 中 Concat axis 没改啊
> > > 0x6 手工微调模型 这一段再看看
> > 
> > 
> > 我刚刚核对的时候也发现，感谢官方答复，现在结果正常了，谢谢！
> 
> 老哥，我跑这个yolact.cpp，模型都能跑出结果，但是后面报错了。请问下你用的什么设备啊？

我先在windows上测得，结果正常了；后来放到了安卓上和linux服务器上，也都没问题

**Comment by gujingxiao** [comment]
官方大佬的教程还是非常棒的，我后来又修改了mobilenetv2做backbone，还修改了各种比例和特征层，都完美支持，手动点个赞。 @nihui

**Comment by ReverseSystem001** [comment]
> > Input layer input.1 without shape info, shape_inference aborted
> > unsupported resize type 0
> > 这两种提示可以无视的...
> 
> 忽略这个进行./ncnn2table会出现这个问题。。
> ./ncnn2table --param ../psenet_lite_nobn_fp32.param --bin ../psenet_lite_nobn_fp32.bin --images ../../../../data/final_test/imgs/ --output psenet_lite_nobn.table --mean=104,117,123 --norm=0.017,0.017,0.017 --size=185,40 --thread=2
> --- ncnn post training quantization tool --- 16:26:39 Apr 26 2020
> fopen true failed
> fopen true failed
> not found [Input] Layer, Check your ncnn.param

我的也有这个问题，但是好像没有太大关系，可能主要是onnx对split等有些不太支持，需要自己微调；我的在load_param((unsigned char*)pParams);时直接报fatal signal 11的错误；
opt.param内容是这样的
7767517
356 428
Input                    input.1                  0 1 input.1
Convolution              Conv_0                   1 1 input.1 399 0=16 1=3 3=2 4=1 5=1 6=432 9=1
Split                    splitncnn_0              1 2 399 399_splitncnn_0 399_splitncnn_1
ConvolutionDepthWise     Conv_3                   1 1 399_splitncnn_1 402 0=16 1=3 4=1 5=1 6=144 7=16 9=1
Convolution              Conv_6                   1 1 402 404 0=16 1=1 5=1 6=256
BinaryOp                 Add_8                    2 1 404 399_splitncnn_0 405
Slice                    Split_9                  1 2 405 input.7 input.8 -23300=2,8,-233
Convolution              Conv_10                  1 1 input.7 408 0=48 1=1 6=384
Convolution              Conv_11                  1 1 input.8 409 0=48 1=1 6=384
Concat                   Concat_12                2 1 408 409 410
BatchNorm                BatchNormalization_13    1 1 410 411 0=96
ReLU                     Relu_14                  1 1 411 412
ConvolutionDepthWise     Conv_15                  1 1 412 415 0=96 1=3 3=2 4=1 5=1 6=864 7=96 9=1
Slice                    Split_18                 1 2 415 input.12 input.13 -23300=2,48,-233
Convolution              Conv_19                  1 1 input.12 418 0=12 1=1 6=576
Convolution              Conv_20                  1 1 input.13 419 0=12 1=1 6=576
Concat                   Concat_21                2 1 418 419 420
BatchNorm                BatchNormalization_22    1 1 420 421 0=24
Split                    splitncnn_1              1 2 421 421_splitncnn_0 421_splitncnn_1
Slice                    Split_23                 1 2 421_splitncnn_1 input.15 input.16 -23300=2,12,-233
Convolution              Conv_24                  1 1 input.15 424 0=36 1=1 6=432
Convolution              Conv_25                  1 1 input.16 425 0=36 1=1 6=432
Concat                   Concat_26                2 1 424 425 426
BatchNorm                BatchNormalization_27    1 1 426 427 0=72
ReLU                     Relu_28                  1 1 427 428
ConvolutionDepthWise     Conv_29                  1 1 428 431 0=72 1=3 4=1 5=1 6=648 7=72 9=1
Slice                    Split_32                 1 2 431 input.20 input.21 -23300=2,36,-233
Convolution              Conv_33                  1 1 input.20 434 0=12 1=1 6=432
Convolution              Conv_34                  1 1 input.21 435 0=12 1=1 6=432
Concat                   Concat_35                2 1 434 435 436
BatchNorm                BatchNormalization_36    1 1 436 437 0=24
BinaryOp                 Add_37                   2 1 437 421_splitncnn_0 438
Convolution              Conv_38                  1 1 438 440 0=144 1=1 5=1 6=3456
Swish                    Mul_41                   1 1 440 442
Slice                    Split_42                 1 3 442 input.25 input.26 input.27 -23300=3,48,48,-233
ConvolutionDepthWise     Conv_43                  1 1 input.25 446 0=48 1=3 3=2 4=1 6=432 7=48
ConvolutionDepthWise     Conv_44                  1 1 input.26 447 0=48 1=5 3=2 4=2 6=1200 7=48
ConvolutionDepthWise     Conv_45                  1 1 input.27 448 0=48 1=7 3=2 4=3 6=2352 7=48
Concat                   Concat_46                3 1 446 447 448 449
BatchNorm                BatchNormalization_47    1 1 449 450 0=144
Swish                    Mul_49                   1 1 450 452
Split                    splitncnn_2              1 2 452 452_splitncnn_0 452_splitncnn_1
Reshape                  Reshape_59               1 1 452_splitncnn_1 464 0=-1 1=144
Reduction                ReduceMean_60            1 1 464 465 0=3 1=0 -23303=1,-1
Reshape                  Reshape_70               1 1 465 479 0=1 1=1 2=144
Convolution              Conv_71                  1 1 479 480 0=12 1=1 5=1 6=1728
Swish                    Mul_73                   1 1 480 482
Convolution              Conv_74                  1 1 482 484 0=144 1=1 5=1 6=1728 9=4
--------------------------------------------------------------------------------------------------------------------------------
又有split，又有swish，啊。。。

**Comment by ReverseSystem001** [comment]
> > Input layer input.1 without shape info, shape_inference aborted
> > unsupported resize type 0
> > 这两种提示可以无视的...
> 
> 忽略这个进行./ncnn2table会出现这个问题。。
> ./ncnn2table --param ../psenet_lite_nobn_fp32.param --bin ../psenet_lite_nobn_fp32.bin --images ../../../../data/final_test/imgs/ --output psenet_lite_nobn.table --mean=104,117,123 --norm=0.017,0.017,0.017 --size=185,40 --thread=2
> --- ncnn post training quantization tool --- 16:26:39 Apr 26 2020
> fopen true failed
> fopen true failed
> not found [Input] Layer, Check your ncnn.param

Hi， brother！I have the same questions. Do you resolve it know???
Hope your reply!

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
